### PR TITLE
Add docs tutorial

### PR DESCRIPTION
**Pull request by zacespinosa**
Added api reference auto docs, fixed up conf.py for sphinx and finished bell state tutorial. check it out and if you see grammar mistakes / I sound extra stupid or confusing please comment :), gracias!

### CODE REVIEW

File: qnetdes/connections.py [hunk 1]
```diff
@@ -31,9 +30,8 @@ def __init__(self, agent_one, agent_two, transit_devices=[]):
             agent_one_name: agent_one.target_devices,
             agent_two_name: agent_two.target_devices,
         }
-        ''' 
-            Assumed order of transit_devices is agent_one -> agent_two (i.e. source -> target).
-        '''
+        
+        # Assumed order of transit_devices is agent_one -> agent_two (i.e. source -> target).
```
Comment (mattradz) [inline qnetdes/connections.py]: Order does not matter though correct?

File: qnetdes/connections.py [hunk 2]
```diff
@@ -48,19 +46,17 @@ def __init__(self, agent_one, agent_two, transit_devices=[]):
             agent_two_name: agent_two
         }
 
-        '''
-            Create queue to keep track of multiple requests. Name of queue is name of
-            target agent.  
-        '''
+        # create queue to keep track of multiple requests. name of queue is name of target agent.
```
Comment (mattradz) [inline qnetdes/connections.py]: "." to ";". Or just capitalize name

File: docs/getting-started.rst [hunk 3]
```diff
@@ -40,36 +40,38 @@ To instead install NetQuil from the source, clone this repository.
...
 
-Import Dependencies 
-----------------------------------------
-.. code:: python
+Import Dependencies
+===================
+Lets start by importing all of our dependencies. 
+
+.. code-block:: python
+    :linenos:
```
Comment (mattradz) [inline docs/getting-started.rst]: I do not see line numbers in the preview here. Also, I will add this to my demos.

File: qnetdes/connections.py (hunk 1 above)
Comment (mattradz) [inline qnetdes/connections.py]: Order does not matter though right?

File: qnetdes/connections.py [hunk 4]
```diff
@@ -9,15 +9,14 @@
...
-            Get devices from source node (e.g. laser, intensity modulator, etc...), 
-            target node (e.g. sensor, beam splitter), and transit devices (e.g. fiber optics, free space)
+        This is the base class for a quantum connection between two agents. 
 
-            :param: Agent agent_one: first agent in connection
-            :param: Agent agent_two: second agent in connection
-            :param  Array transit_devices: array of devices packets travel through - assumed order: agent_one -> agent_two
+        :param Agent agent_one: first agent in connection
+        :param Agent agent_two: second agent in connection
+        :param Array transit_devices: array of devices packets travel through - assumed order: agent_one -> agent_two
```
Comment (mattradz) [inline qnetdes/connections.py]: "packets" to "qubits

File: qnetdes/agents.py [hunk 5]
```diff
@@ -92,23 +107,25 @@ def _tracer(self, frame, event, arg):
...
 
             :param List new_target_devices: Agent new target devices
         '''
         self.target_devices.extend(new_target_devices)
 
     def add_source_devices(self, new_source_devices):
         '''
-            Add self source devices
+            Add source devices. Every qubit sent to Agent will pass through these devices, 
+            and, if set, the device's noise will be applied
```
Comment (mattradz) [inline qnetdes/agents.py]: These descriptions are reversed right? Target devices act on qubits that sent TO the Agent and Source devices act on qubits that are sent FROM the Agent.

File: demos/non-local-cnot.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            self.crecv(bob.name) 
+            p.if_then(ro[1], Z(0))
+            p += MEASURE(0, ro[0])
+
+    def cat_entangler(self, p):
+        p += CNOT(2,0)
+        p += MEASURE(0, ro[0])
+        p.if_then(ro[0], X(0))
+        
+        self.csend(bob.name, [0])
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply demos/non-local-cnot.py]: yeah, i can remove them if they're confusing or we could add cmem to tracer and force people to csend before trying to do ro. What do you think?

File: docs/api-reference/distributed-gates.rst [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.. _distributed-gates:
+
+``Distributed Quantum Gates``
+----------------------------------
+.. automodule:: qnetdes.distributed-gates
+   :members:
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply docs/api-reference/distributed-gates.rst]: idk if we need all three in all places. usage is sorta explained here http://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-automodule

[the remark this answers is not in the document]
Comment (zacespinosa) [reply docs/api-reference/distributed-gates.rst]: I'll check and see what I can get rid of

File: docs/conf.py [hunk 8]
```diff
@@ -29,8 +33,20 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = [
-]
+extensions = ['sphinx.ext.autodoc',
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply docs/conf.py]: the automodule stuff for api-reference

File: demos/non-local-cnot.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        
+        p += MEASURE(1, ro[1])
+
+    def run(self):
+        p = self.program
+        self.get_entangle_pair(p)
+        self.cat_entangler(p)
+        self.cat_disentangler(p)
+
+wf_sim = WavefunctionSimulator()
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply demos/non-local-cnot.py]: sounds good,  i'll change to printWF

File: demos/non-local-cnot.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+import sys
+sys.path.insert(0, '/Users/zacespinosa/Foundry/qnetdes')
+sys.path.insert(1, '/Users/matthewradzihovsky/documents/qnetdes')
+
+from pyquil import Program
+from pyquil.api import WavefunctionSimulator, QVMConnection
+from pyquil.gates import *
+from qnetdes import *
+
+class Alice(Agent):
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply demos/non-local-cnot.py]: sg, non-local-cnot is still WIP

File: docs/getting-started.rst (hunk 3 above)
Comment (zacespinosa) [reply docs/getting-started.rst]: yeah, the git preview isn't perfect :(

File: docs/getting-started.rst [hunk 11]
```diff
@@ -120,17 +124,23 @@ from one agent to another.
             self.log_results(r)
```
[the remark this answers is not in the document]
Comment (zacespinosa) [reply docs/getting-started.rst]: not defined, imma remove for now and do 'print(r)'

File: qnetdes/connections.py (hunk 1 above)
Comment (zacespinosa) [reply qnetdes/connections.py]: Order matters if the devices introduce noise right?

**Pull request merged**